### ISSUE TITLE
Support for option types?

### ISSUE THREAD

**Issue by isaacabraham** [fsprojects/ExcelProvider#31]
### Description

I can't find any support for optional data within the TP - so I'm forced to fall back to working with strings and then converting to e.g. floats.

Have I missed something?

**Comment by giuliohome** [comment]
See also my workaround https://github.com/fsprojects/ExcelProvider/issues/56#issuecomment-582830687
If the maintainer consider this an enhancement that is not even planned for implementation, they should clearly write this project is experimental and not ready for production...

**Comment by abelbraaksma** [comment]
@giuliohome, since this is open source and I believe it started as a proof of concept for type providers, it's a bit rough on the edges. What this project really needs is some contributors that like to step in. I'm sure you can write a PR (the source of this project isn't that complex).

And personally there are some features I might contribute to myself, for instance the dynamic range selection, or adding discovery of sheetnames. It'd be nice to work together and get some open issues resolved.

**Comment by giuliohome** [comment]
Yes, after a quick look at the source I would guess that the point could be around [here](https://github.com/fsprojects/ExcelProvider/blob/master/src/ExcelProvider.DesignTime/ExcelProvider.DesignTime.fs#L115). 

Well, maybe I can write a PR, I'll think about it...

**Comment by giuliohome** [comment]
Or maybe it's better to discuss the idea before writing any code.

It seems it look at the first row to determine types. One could assume that the template is chosen and modified in order to simplify type inferring, so the easiest idea could be: let's guess the type with the first two rows instead of only one. And let's say that the inferred type is T option when one cell is empty and the other has inferred type T.

Hopefully, this could be enough from my side either if someone else is more ready than me to do the actual code change or if someone else already disagrees with the proposed idea.

**Comment by quintusm** [comment]
HI all.
I am in the process of doing an update to move the code base to FAKE5 and .netstandard only as per the template for type providers. This has proven to take much longer than anticipated and I have had less time to spend on it than I expected.  

Once that is done it should be easier to make progress on the core functionality of the project and I will be looking at trying to implement more of the outstanding requests. There is a lot that could be done.  In the meantime of course any contributions would be most welcome.
